src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -244,8 +245,16 @@ void Run (bool useMarshalMethods)
 
 				foreach (var kvp in nativeCodeGenStates) {
 					NativeCodeGenState state = kvp.Value;
-					RewriteMarshalMethods (state, brokenExceptionTransitionsEnabled);
-					state.Classifier.AddSpecialCaseMethods ();
+					if (!EnableManagedMarshalMethodsLookup) {
+						RewriteMarshalMethods (state, brokenExceptionTransitionsEnabled);
+						state.Classifier.AddSpecialCaseMethods ();
+					} else {
+						// We need to run `AddSpecialCaseMethods` before `RewriteMarshalMethods` so that we can see the special case
+						// methods (such as TypeManager.n_Activate_mm) when generating the managed lookup tables.
+						state.Classifier.AddSpecialCaseMethods ();
+						state.ManagedMarshalMethodsLookupInfo = new ManagedMarshalMethodsLookupInfo (Log);
+						RewriteMarshalMethods (state, brokenExceptionTransitionsEnabled);
+					}
```

Comment:
I did not observe any functional changes when swapping the order of `AddSpecialCaseMethods` and `RewriteMarshalMethods`. I also don't see a reason that the code in `AddSpecialCaseMethods` could not run before `RewriteMarshalMethods`. Nevertheless, I thought it might not be a good idea to change the existing behavior and only change the order when this (experimental) feature is enabled.

Reply:
That's what CI is for: a safety net.  Try it and see! ;-)

We can certainly leave this for now.